nixos/modules/services/security/proxychains.nix

```diff
@@ -0,0 +1,184 @@
+          description = ''
+            <literal>dynamic</literal> - Each connection will be done via chained proxies
+            all proxies chained in the order as they appear in the list
+            at least one proxy must be online to play in chain
+            (dead proxies are skipped)
+            otherwise <literal>EINTR</literal> is returned to the app.
+
+            <literal>strict</literal> - Each connection will be done via chained proxies
+            all proxies chained in the order as they appear in the list
+            all proxies must be online to play in chain
+            otherwise <literal>EINTR</literal> is returned to the app.
+
+            <literal>random</literal> - Each connection will be done via random proxy
+            (or proxy chain, see <option>services.proxychains.chain.length</option>) from the list.
+          '';
+        };
+        length = mkOption {
+          type = types.nullOr types.int;
+          default = null;
+          description = ''
+            Chain length for random chain.
+          '';
+        };
+      };
+
+      proxyDNS = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Proxy DNS requests - no leak for DNS data.";
+      };
+
+      quietMode = mkEnableOption "Quiet mode (no output from the library).";
+
+      remoteDNSSubnet = mkOption {
+        type = types.enum [ 10 127 224 ];
+        default = 224;
+        apply = x: "remote_dns_subnet ${builtins.toString x}";
+        description = ''
+          Set the class A subnet number to use for the internal remote DNS mapping, uses the reserved 224.x.x.x range by default.
+        '';
+      };
+
+      tcpReadTimeOut = mkOption {
+        type = types.int;
+        default = 15000;
+        description = "Connection read time-out in milliseconds.";
+        apply = x: "tcp_read_time_out ${builtins.toString x}";
+      };
+
+      tcpConnectTimeOut = mkOption {
+        type = types.int;
+        default = 8000;
+        description = "Connection time-out in milliseconds.";
+        apply = x: "tcp_connect_time_out ${builtins.toString x}";
+      };
+
+      localnet = mkOption {
+        type = types.str;
+        default = "127.0.0.0/255.0.0.0";
+        description = "By default enable localnet for loopback address ranges.";
+      };
+
+      proxies = mkOption {
+        type = types.listOf (types.submodule proxyOptions);
+        description = ''
+          List of proxies to be used by proxychains.
+        '';
+
+        example = literalExample ''
+          [
+            { type = "socks4";
+              host = "127.0.0.1";
+              port = 1337;
+            }
+          ]
+        '';
+
+        apply = xs:
+          let proxyLine = x: "${x.type} ${x.host} ${x.port}";
+          in ''
+            [ProxyList]
+            ${builtins.concatStringsSep "\n" (builtins.map proxyLine xs)}
+          '';
+      };
+
+      extraConfig = mkOption {
```

Comment:
Ideally such an option wouldn't be added unless there's a good reason for it, see also NixOS/rfcs#42